### PR TITLE
(QA-2458) rototiller  document document document

### PR DESCRIPTION
**Pull request by er0ck**
[skip ci]

### CODE REVIEW

**Comment by er0ck** [comment]
removed class structure: too difficult to maintain and we have yardocs.  not user doc related.

File: README.md [hunk 1]
```diff
@@ -1,61 +1,143 @@
...
+
+    $) rake test_arg_env
+    Rakefile
+
+    $) rake test_arg_env FILENAME=README.md
+    README.md
+
+## Issues
+
+* none. it's perfect
```
Comment (zreichert) [inline README.md:122]: Mostly true 👍

File: README.md [hunk 2]
```diff
@@ -1,61 +1,143 @@
...
+    $) rake test_flag_env --verbose FLAG_VALUE='nonesuch'
+    The CLI flag -f will be used with value README.md.
+
+    test -f README.md
+    test -f nonesuch failed
+
+    $) rake test_arg_env
+    Rakefile
+
+    $) rake test_arg_env FILENAME=README.md
```
Comment (er0ck) [inline README.md:117]: no env_var messaging.  ticketed seperately

File: README.md [hunk 3]
```diff
@@ -1,61 +1,143 @@
...
+    The CLI flag -f will be used with value Rakefile.
+    $) echo $?
+    0
+
+    $) rake test_flag_env --verbose
+    The CLI flag -f will be used with value Rakefile.
+
+    test -f Rakefile
+
+    $) rake test_flag_env --verbose FLAG_VALUE='README.md'
```
Comment (er0ck) [inline README.md:103]: no messaging on where the value came from (what env, default or not). ticketed separately

File: README.md [hunk 4]
```diff
@@ -1,61 +1,143 @@
...
+    desc "override flag values with environment variables"
+    rototiller_task :test_flag_env do |task|
+      task.add_command do |cmd|
+        cmd.name = 'test'
+      end
+      task.add_flag do |flag|
+        flag.name         = '-f'
+        flag.default      = 'Rakefile'
+        flag.override_env = 'FLAG_VALUE'
+      end
```
Comment (er0ck) [inline README.md:48]: add_flag should be method in command, not task.  ticketed separately

**Pull request merged**